projects/kalidea/kaligraphi/src/lib/atoms/kal-slider/kal-slider.component.ts

```diff
@@ -22,192 +21,183 @@ import { buildProviders, FormControlAccessComponent } from '../../utils/index';
-  set disabled(isDisabled: boolean) {
-    this.isDisabled = isDisabled;
-    this.toggle();
+  get sliderDimensions(): ClientRect | null {
+    return this.sliderWrapper ? this.sliderWrapper.nativeElement.getBoundingClientRect() : null;
   }
 
-  /**
-   * Whether the slider should be disabled.
-   */
-  @Input()
-  get step(): number {
-    return this.sliderStep;
-  }
-  set step(step: number) {
-    this.sliderStep = step;
-    this.cdr.markForCheck();
+  @HostListener('mousedown', ['$event'])
+  mouseDown($event: MouseEvent) {
+    if (this.disabled) {
+      return;
+    }
+    const offset = this.sliderDimensions.left;
+    const position = $event.clientX - offset;
+    this.value = this.positionInSliderToValue(position);
   }
 
-  /** Whether or not to show the thumb label. */
-  @Input()
-  get thumbLabel(): boolean {
-    return this.displayThumbLabel;
-  }
-  set thumbLabel(value: boolean) {
-    this.displayThumbLabel = coerceBooleanProperty(value);
-    this.cdr.markForCheck();
+  @HostListener('keydown', ['$event'])
+  keyDown($event: KeyboardEvent) {
+    if (this.disabled) {
+      return;
+    }
+    const numSteps = $event.ctrlKey ? 10 : 1;
+    switch ($event.keyCode) {
+      case HOME:
+        this.value = this.minValue;
+        break;
+      case END:
+        this.value = this.maxValue;
+        break;
+      case LEFT_ARROW:
+        this.increment(-numSteps);
+        break;
+      case RIGHT_ARROW:
+        this.increment(numSteps);
+        break;
+    }
   }
 
-  /**
-   * How often to show ticks. Relative to the step so that a tick always appears on a step.
-   * Ex: Tick interval of 4 with a step of 3 will draw a tick every 4 steps (every 12 values).
-   */
-  @Input()
-  get tickInterval(): number {
-    return this.tickIntervalValue;
-  }
-  set tickInterval(value: number) {
-    this.tickIntervalValue = coerceNumberProperty(value);
-    this.cdr.markForCheck();
-  }
+  @HostListener('slide', ['$event'])
+  slide($event: HammerInput) {
 
-  /**
-   * @inheritDoc
-   */
-  writeValue(newValue: number): void {
-    if (newValue !== this.value) {
-      // get the limit value
-      this.value = this.getLimitValue(newValue);
-      this.control.patchValue(this.value, {emitEvent: false});
+    if (this.disabled) {
+      return;
+    }
 
-      // notify parent
-      super.writeValue(this.value);
+    // Prevent the slide from selecting anything else.
+    $event.preventDefault();
 
-      // update the view when we get a new value from parent component
-      this.cdr.markForCheck();
-    }
+    const offset = this.sliderDimensions.left;
+    const position = $event.center.x - offset;
+
+    this.value = this.positionInSliderToValue(position);
   }
 
-  /**
-   * @inheritDoc
-   */
-  setDisabledState(isDisabled: boolean): void {
-    this.isDisabled = isDisabled;
+  selectionContainerStyles(): { 'width.%': number, backgroundColor?: string } {
+    const styles: { 'width.%': number, backgroundColor?: string } = {
```

Comment:
const styles: { 'width.%': number, backgroundColor?: string } = {
     'width.%': this.valueToPercent(this.value),
     backgroundColor = this.color || undefined
}